tls/s2n_config.c

```diff
@@ -451,6 +451,13 @@ int s2n_config_set_check_stapled_ocsp_response(struct s2n_config *config, uint8_
     return 0;
 }
 
+int s2n_config_disable_x509_validity_period_validation(struct s2n_config *config)
+{
+    POSIX_ENSURE_REF(config);
+    config->disable_x509_validity_period_validation = true;
```

Comment:
```suggestion
    config->ignore_x509_validity_period = true;
```
Just thinking of better names... otherwise this flag is just so long.

Reply:
Are we attached to "validity_period"? Maybe "disable_x509_time_checks"? "disable_x509_timestamp_checks"?

Reply:
Validity period is what the RFC calls this, but we definitely don't have to use that. I renamed the API to `s2n_config_validate_x509_time()`: https://github.com/aws/s2n-tls/pull/4183#discussion_r1317597722